broadcaster/lib/broadcaster_web/channels/stream_channel.ex

```diff
@@ -11,22 +11,39 @@ defmodule BroadcasterWeb.StreamChannel do
     {:ok, assign(socket, :nickname, nil)}
   end
 
+  @impl true
+  def join("stream:chat-admin", _, socket) do
+    {:ok, socket}
+  end
+
   @impl true
   def handle_in("chat_msg", _, %{assigns: %{nickname: nil}} = socket) do
     {:noreply, socket}
   end
 
   @impl true
   def handle_in("chat_msg", %{"body" => body}, socket) do
-    broadcast!(socket, "chat_msg", %{body: body, nickname: socket.assigns.nickname})
-    {:noreply, socket}
+    msg = %{
+      body: body,
+      nickname: socket.assigns.nickname,
+      id: "#{socket.assigns.nickname}:#{socket.assigns.id}"
+    }
+
+    broadcast!(socket, "chat_msg", msg)
+    BroadcasterWeb.Endpoint.broadcast!("stream:chat-admin", "chat_msg", msg)
+
+    {:noreply, assign(socket, :id, socket.assigns.id + 1)}
   end
 
   @impl true
   def handle_in("join_chat", %{"nickname" => nickname}, socket) do
     case register(nickname) do
       :ok ->
-        socket = assign(socket, :nickname, nickname)
+        socket =
+          socket
+          |> assign(:nickname, nickname)
+          |> assign(:id, 0)
```

Comment:
Better start with a random value

Reply:
Instead, I'll use completely unique id to create the message id.

Reply:
Nitpick:

```suggestion
          |> assign(:msg_cnt, 0)
```